src/modding/NewModGUI.cs

```diff
@@ -217,6 +300,57 @@ private bool ReadControlsToEditedInfo()
         editedInfo.ModAssembly = modAssembly.Text;
         editedInfo.AssemblyModClass = assemblyModClass.Text;
 
+        if (!string.IsNullOrWhiteSpace(previewImagesFile.Text))
+        {
+            var previewImagesList = new List<string>();
+            Array.ForEach(previewImagesFile.Text.Split(","), s => previewImagesList.Add(s.Trim()));
+            editedInfo.PreviewImages = previewImagesList;
+        }
+
+        if (!string.IsNullOrWhiteSpace(dependencies.Text))
+        {
+            var dependenciesList = new List<string>();
+            Array.ForEach(dependencies.Text.Split(","), s => dependenciesList.Add(s.Trim()));
+            editedInfo.Dependencies = dependenciesList;
+        }
+
+        if (!string.IsNullOrWhiteSpace(requiredMods.Text))
+        {
+            var requiredModsList = new List<string>();
+            Array.ForEach(requiredMods.Text.Split(","), s => requiredModsList.Add(s.Trim()));
+            editedInfo.RequiredMods = requiredModsList;
+        }
+
+        if (!string.IsNullOrWhiteSpace(loadBefore.Text))
+        {
+            var loadBeforeList = new List<string>();
+            Array.ForEach(loadBefore.Text.Split(","), s => loadBeforeList.Add(s.Trim()));
+            editedInfo.LoadBefore = loadBeforeList;
+        }
+
+        if (!string.IsNullOrWhiteSpace(loadAfter.Text))
+        {
+            var loadAfterList = new List<string>();
+            Array.ForEach(loadAfter.Text.Split(","), s => loadAfterList.Add(s.Trim()));
+            editedInfo.LoadAfter = loadAfterList;
+        }
+
+        if (!string.IsNullOrWhiteSpace(incompatibleMods.Text))
+        {
+            var incompatibleModsList = new List<string>();
+            Array.ForEach(incompatibleMods.Text.Split(","), s => incompatibleModsList.Add(s.Trim()));
+            editedInfo.IncompatibleMods = incompatibleModsList;
+        }
+
+        if (enableConfigCheckbox.Pressed && !string.IsNullOrWhiteSpace(modConfig.Text))
+        {
+            editedInfo.ConfigToLoad = modConfig.Text;
+            if (!modConfig.Text.EndsWith(".json"))
+            {
+                editedInfo.ConfigToLoad += ".json";
```

Comment:
Ok, so under the condition, `modConfig.Text` is precisely `editedInfo.ConfigToLoad` and what you really test is the latter, right? This should be explicit.

Btw, I foresee a potential helper `ForceExtension` method to do this, as it would have a great reusability potential !

Reply:
Yeah i want to make sure that it ends in a json extension if the user did not end it in a json extension themselves

Reply:
I think it would be better to disallow entering incorrect data to begin with... see how I made the mod uploader require valid data before the upload button can be pressed.